src/fast_auth_token/mod_fast_auth_token_generic_mech.erl

```diff
@@ -64,6 +70,29 @@ mech_step(#fast_info{creds = Creds, agent_id = AgentId, mechanism = Mech, count
             {error, <<"not-authorized">>}
     end.
 
+mech_to_cb_type(Mech) ->
+    Type = lists:last(binary:split(Mech, <<"-">>, [global])),
+    cb_type_to_atom(Type).
+
+cb_type_to_atom(<<"NONE">>) -> none;
+cb_type_to_atom(<<"EXPR">>) -> expr.
+
+-spec mech_to_cb_data(
+    Mech :: mod_fast_auth_token:mechanism(),
+    Socket :: mongoose_xmpp_socket:socket()) -> cb_data().
+mech_to_cb_data(Mech, Socket) ->
+    case mech_to_cb_type(Mech) of
+        none ->
+            <<>>;
+        expr ->
+            case mongoose_xmpp_socket:export_key_materials(
+                    Socket, [?CB_LABEL], [no_context], [32], true) of
```

Comment:
A lot of this input is a bit magic numbers, so please explain them well, put comments or so :)

Reply:
added :)